pennylane/ops/op_math/controlled.py

```diff
@@ -551,11 +554,23 @@ def _decompose_no_control_values(op: "operation.Operator") -> List["operation.Op
     no decomposition.
     """
     if len(op.control_wires) == 1 and hasattr(op.base, "_controlled"):
-        return [op.base._controlled(op.control_wires[0])]
+        result = op.base._controlled(op.control_wires[0])
+        # disallow decomposing to itself
+        # pylint: disable=unidiomatic-typecheck
+        if type(result) != type(op):
+            return [result]
+        qml.QueuingManager.remove(result)
     if isinstance(op.base, qml.PauliX):
-        if len(op.control_wires) == 2:
-            return [qml.Toffoli(op.active_wires)]
```

Comment:
Why are we getting rid of the `Toffoli` case? Yes, `MultiControlledX` will decompose into a `Toffoli` itself for this case, but `Toffoli` is a fairly important case we don't want to just bypass.

Reply:
the linter complained there were too many return statements, it was harmless to get rid of this one

Reply:
the max is 6 btw, there are now 6, if you add this back there will be 7 and this PR will not be able to be merged

Reply:
got the ok on this

Reply:
For the future, you can always use `# pylint: disable=[insert-lint-errors-here]`. This is an instance where having "too many" return statements is justified, so I don't mind having to silence pylint. I'll unresolve this conversation for now, but if you've gotten a green light for skipping on the `Toffoli` case, you can just mark it as resolved again.

Reply:
codefactor would complain about it too. and yes christina said it's ok